x-pack/plugins/observability/public/pages/rules/rules.tsx

```diff
@@ -48,42 +62,65 @@ export function RulesPage() {
     useHashQuery: false,
   });
 
-  const { status, lastResponse } = urlStateStorage.get<{
-    status: RuleStatus[];
+  const { lastResponse, search, status, type } = urlStateStorage.get<{
     lastResponse: string[];
-  }>('_a') || { status: [], lastResponse: [] };
+    search: string;
+    status: RuleStatus[];
+    type: string[];
+  }>('_a') || { lastResponse: [], search: '', status: [], type: [] };
 
-  const [stateStatus, setStatus] = useState<RuleStatus[]>(status);
   const [stateLastResponse, setLastResponse] = useState<string[]>(lastResponse);
+  const [stateSearch, setSearch] = useState<string>(search);
+  const [stateStatus, setStatus] = useState<RuleStatus[]>(status);
+  const [stateType, setType] = useState<string[]>(type);
+
   const [stateRefresh, setRefresh] = useState(new Date());
 
   const [addRuleFlyoutVisibility, setAddRuleFlyoutVisibility] = useState(false);
 
-  useBreadcrumbs([
-    {
-      text: i18n.translate('xpack.observability.breadcrumbs.alertsLinkText', {
-        defaultMessage: 'Alerts',
-      }),
-      href: http.basePath.prepend('/app/observability/alerts'),
-    },
-    {
-      text: i18n.translate('xpack.observability.breadcrumbs.rulesLinkText', {
-        defaultMessage: 'Rules',
-      }),
-    },
-  ]);
-
   const handleStatusFilterChange = (newStatus: RuleStatus[]) => {
     setStatus(newStatus);
-    urlStateStorage.set('_a', { status: newStatus, lastResponse });
-    return { lastResponse: stateLastResponse || [], status: newStatus };
+    urlStateStorage.set('_a', { lastResponse, search, status: newStatus, type });
+    return {
+      lastResponse: stateLastResponse || [],
+      search: search || '',
+      status: newStatus,
+      type: stateType,
```

Comment:
It looks like we fallback on empty array below
```suggestion
      type: stateType || [],
```